Controllers/PostController.cs

```diff
@@ -86,4 +86,35 @@ public IActionResult CreatePost(Post post)
         _dbContext.SaveChanges();
         return Created($"/api/post/{post.Id}", post);
     }
+
+    [HttpDelete("{postId}")]
+    [Authorize]
+    public IActionResult DeletePost( int postId)
+    {
+        Post PostToDelete = _dbContext.Posts.SingleOrDefault(p => p.Id == postId);
+        if (PostToDelete == null)
+        {
+            return NotFound();
+        }
+        else if (postId != PostToDelete.Id)
+        {
+            return BadRequest();
+        }
+        //delete comment for post
+        var relatedComments = _dbContext.Comments.Where(c => c.PostId == postId);
+        _dbContext.Comments.RemoveRange(relatedComments);
+
+        //delete reactions for post
+        var relatedReactions = _dbContext.PostReactions.Where(pr => pr.PostId == postId);
+        _dbContext.PostReactions.RemoveRange(relatedReactions);
+
+        //delete post tags
+        var relatedTags = _dbContext.PostTags.Where(pt => pt.PostId == postId);
+        _dbContext.PostTags.RemoveRange(relatedTags);
+
```

Comment:
Your code handles cascading deletions for related comments, reactions, and tags, which is a good practice to ensure data consistency. 
However I need a second pair of eyes to look at this, I'm not sure if our database relationships will automatically handle cascading deletions or not.